app/workers/downstream_publish_worker.rb

```diff
@@ -0,0 +1,93 @@
+class DownstreamPublishWorker
+  attr_reader :content_item_id, :payload_version, :message_queue_update_type, :update_dependencies
+  include Sidekiq::Worker
+  include PerformAsyncInQueue
+
+  HIGH_QUEUE = "downstream_high".freeze
+  LOW_QUEUE = "downstream_low".freeze
+
+  sidekiq_options queue: HIGH_QUEUE
+
+  def perform(args = {})
+    assign_attributes(args.symbolize_keys)
+
+    unless web_content_item
+      raise CommandError.new(
+        code: 404,
+        message: "The content item for id: #{content_item_id} was not found",
+      )
+    end
+
+    unless content_item_state?(:published)
+      raise CommandError.new(
+        code: 500,
+        message: "Will not downstream publish a content item that isn't published",
+      )
+    end
+
+    send_to_live_content_store if should_send_to_content_store?
+    enqueue_dependencies if update_dependencies
+    broadcast_to_message_queue
+  end
+
+private
+
+  def assign_attributes(attributes)
+    @content_item_id = attributes.fetch(:content_item_id)
+    @payload_version = attributes.fetch(:payload_version)
+    @message_queue_update_type = attributes.fetch(:message_queue_update_type)
+    @update_dependencies = attributes.fetch(:update_dependencies, true)
+  end
+
+  def send_to_live_content_store
+    payload = presented_content_store_payload
+    base_path = payload.fetch(:base_path)
```

Comment:
sure, yeah I actually switched to that on one of my other workers. I'll modify this one accordingly.